task/upload.go

```diff
@@ -72,27 +74,57 @@ func handleUploadVOD(p handleUploadVODParams) (*TaskHandlerOutput, error) {
 		if err != nil {
 			return nil, err
 		}
-		var encryption *clients.EncryptionPayload
-		params := tctx.Task.Params.Upload
 
-		if params != nil && params.Encryption.EncryptedKey != "" {
-			encryption = &clients.EncryptionPayload{
-				EncryptedKey: params.Encryption.EncryptedKey,
+		var encryption *clients.EncryptionPayload
+		var clipStrategy *video.ClipStrategy
+
+		if tctx.Task.Params.Clip != nil {
+			clipParams := tctx.Task.Params.Clip
+			if clipParams != nil {
+				clipStrategy = &video.ClipStrategy{
+					StartTime:  clipParams.ClipStrategy.StartTime,
+					EndTime:    clipParams.ClipStrategy.EndTime,
+					PlaybackID: clipParams.ClipStrategy.PlaybackId,
+				}
+			}
+		} else {
+			uploadParams := tctx.Task.Params.Upload
+			if uploadParams != nil && uploadParams.Encryption.EncryptedKey != "" {
+				encryption = &clients.EncryptionPayload{
+					EncryptedKey: uploadParams.Encryption.EncryptedKey,
+				}
 			}
 		}
-		var (
-			req = clients.UploadVODRequest{
-				ExternalID:            tctx.Task.ID,
-				Url:                   inUrl,
-				CallbackUrl:           tctx.catalyst.CatalystHookURL(tctx.Task.ID, "finalize", catalystTaskAttemptID(tctx.Task)),
-				OutputLocations:       outputLocations,
-				PipelineStrategy:      p.catalystPipelineStrategy,
-				Profiles:              p.profiles,
-				TargetSegmentSizeSecs: p.targetSegmentSizeSecs,
-				Encryption:            encryption,
+
+		var req clients.UploadVODRequest
```

Comment:
Maybe at some point we just want to move this `clients.UploadVODRequest` to the `handleUploadVODParams` and let callers create their catalyst requests. Otherwise we have this "generic" "handle upload" function that is not really generic, just has a bunch of conditionals to implement all the possible task types this might run. Might as well not have the abstraction if it grows like that.